pulpcore/cli/container/repository.py

```diff
@@ -165,3 +247,77 @@ def add_tag(
 @pass_repository_context
 def remove_tag(repository_ctx: PulpContainerBaseRepositoryContext, tag: str) -> None:
     repository_ctx.untag(tag)
+
+
+@repository.command(allowed_with_contexts=container_context)
+@name_option
+@href_option
+@click.option(
+    "--source",
+    help=_(
+        "Source repository to copy tags from. Specify in format of href or [<resource_type>:]<name>"
+        "[:<version>]"
+    ),
+    required=True,
+    callback=_source_callback,
+)
+@click.option(
+    "--tag",
+    "tags",
+    help=_("Multiple option of tag names to copy, leave blank to copy all"),
+    multiple=True,
+)
+@pass_repository_context
+def copy_tag(
+    repository_ctx: PulpContainerRepositoryContext,
+    source: PulpRepositoryVersionContext,
+    tags: List[str],
+) -> None:
+    repository_ctx.copy_tag(source_href=source.pulp_href, tags=tags or None)
+
+
+@repository.command(allowed_with_contexts=container_context)
+@name_option
+@href_option
+@click.option(
+    "--source",
+    help=_(
+        "Source repository to copy manifests from. Specify in format of href or [<resource_type>:]"
+        "<name>[:<version>]"
```

Comment:
:dragon: optional parts in two places sounds real dangerous. Also i had expected that wherever we have a resource parameter, we'd allow for `[[<plugin>:]<type>:]<name>`.
Also we need to make sure we still allow ":" in the name of the repository with `::reponame:with:colons`.
Is it very unintuitive to use a separate separator for the version?